packages/grafana-data/src/types/logsVolume.ts

```diff
@@ -0,0 +1,19 @@
+import { DataQuery } from './query';
+import { DataQueryRequest, DataQueryResponse } from './datasource';
+import { Observable } from 'rxjs';
+
+/**
+ * TODO: This should be added to ./logs.ts but because of cross reference between ./datasource.ts and ./logs.ts it can
+ * be done only after decoupling "logs" from "datasource" (https://github.com/grafana/grafana/pull/39536)
+ *
+ * @internal
+ */
+export interface DataSourceWithLogsVolumeSupport<TQuery extends DataQuery> {
+  getLogsVolumeDataProvider(request: DataQueryRequest<TQuery>): Observable<DataQueryResponse> | undefined;
+}
+
+export const hasLogsVolumeSupport = <TQuery extends DataQuery>(
```

Comment:
@aocenas

> When you use scroll navigation/loading we seem to use the cache somehow but if we load more logs we do reset the histogram even though we do not go outside of the histogram range so we could still keep it (or not do a new request if autoloading == true)

Very good point. Do you think it could be added in a separate PR?

Reply:
Yes don't think it is a bit issue at this moment, just something to keep in mind as an optimization.

Reply:
This be worked on in #39856.